Frontend/src/ui/Competition.jsx

```diff
@@ -0,0 +1,24 @@
+import { useQuery } from '@tanstack/react-query'
+import React, { useEffect, useState } from 'react'
+import { useParams } from 'react-router-dom'
+import getCompetitionInfo from '../api/competition/get/get_competition_info'
+import { CompetitionContext } from '../api/helper/context/competition_context'
+
+export default function Competition({ children }) {
+  const { competition_id } = useParams()
+  const { data } = useQuery({
+    queryKey: [competition_id],
+    queryFn: () => getCompetitionInfo(competition_id),
+  })
+  const [competitionInfo, setCompetitionInfo] = useState({})
+  useEffect(() => {
+    if (data) {
+      setCompetitionInfo(data)
+    }
+  }, [data])
```

Comment:
You don't need this state or effect, `useQuery` is handling all this for you. Just do `<CompetitionContext.Provider value={{ competitionInfo: data ?? {} }}>` (and can possibly omit the `?? {}` if also doing my comment on the line below).

Typically you won't need any `useEffect`s when working with react-query.

Reply:
ah okay. I removed it and added a `LoadingMessage`